src/GeoInterfaceRFC.jl

```diff
@@ -5,47 +5,84 @@ struct Geometry <: AbstractGeometry end
 
 abstract type AbstractPoint <: AbstractGeometry end
 struct Point <: AbstractPoint end
-abstract type AbstractLineString <: AbstractGeometry end
+
+abstract type AbstractCurve <: AbstractGeometry end
+abstract type AbstractLineString <: AbstractCurve end
 struct LineString <: AbstractLineString end
-abstract type AbstractPolygon <: AbstractGeometry end
+struct Line <: AbstractLineString end  # LineString with just two points
+struct LinearRing <: AbstractLineString end  # Closed LineString
+
+abstract type AbstractSurface <: AbstractGeometry end
+abstract type AbstractPolygon <: AbstractSurface end
+abstract type AbstractPolyHedralSurface <: AbstractSurface end
 struct Polygon <: AbstractPolygon end
+struct Triangle <: AbstractPolygon end  # Polygon with just three points
+struct Box <: AbstractPolygon end  # Polygon with just 4 points
```

Comment:
Hmm this one is not in your picture at https://github.com/JuliaGeo/GeoInterfaceRFC.jl/issues/3. Do you think it's important to add it? And what about naming it `Rectangle`, since `Box` could be interpreted as a 3D structure?

Reply:
I guess this is not necessarily orthogonal to the axis, like the `BoundingBox`, right?

Reply:
I guess this would be a `Quad` in the general sense (if it can have 4 points)?

Reply:
It's not a Quad, but it could be another Special type. The Box is rectangular, while the Quad is more generic and just specifies four corners and can thus also be diamond shaped.

Reply:
I added `Quad` as well.